BE: Adds gmail authentication capabilities to server

cpascale43 opened the pull request:
**What it does**
- Connects our server via OAuth to Gmail for use in #12. 

**What to look for**
- Download the JSON file (`credentials.json`) sent over Slack and import to your local /server/auth directory.

**What to avoid**
- Exposing client secrets in our project directory. `credentials.json` should be in .gitignore, but double check this.

Commit:
init update for database componet

cpascale43 committed:
adds gmail authentication capabilities to server

Commit:
Remove google_account_linked and sent_emails in User model
Freeze library version with pipfile.lock

cpascale43 committed:
Merge branch 'dev' of https://github.com/hatchways/team-autumn into be-gmail-auth-server
access changes to dev # the commit.

Commit:
Freeze library version with pipfile

cpascale43 committed:
Merge branch 'BE_setup_mongo_for_project_19' of https://github.com/hatchways/team-autumn into be-gmail-auth-server
insert changes from patton# the commit.

mcgraths7 commented on server/auth/gmail.py:
```diff
@@ -0,0 +1,53 @@
+from google.auth.transport.requests import Request
+
+# If modifying these scopes, delete the file token.pickle.
+SCOPES = ['https://www.googleapis.com/auth/gmail.readonly']
```
If we want to be able to send mail, would we want a scope which would let us send mail on the user's behalf? (https://www.googleapis.com/auth/gmail.compose)
--

Commit:
Handle case when Team name is not specified

Commit:
add util for common methods access

Commit:
Use jsonschema to santize user input;
Full register logic implemented

Commit:
add register blueprint to app.
add flask extensions to app.
add default value for testing.

Commit:
add alias arg to db

Commit:
Use error code to indicate errors

Commit:
Email is no longer the primary key;
add class method to make User model fit OOP.

Commit:
init update of login handler

Commit:
spelling mistake fix

Commit:
for consistency add error_code 0 for success request

Commit:
add test cases for register_handler

Commit:
user_schema now has correct type.
user input will now be parse directly instead of falsely double dump.
user input will be copyed before delete password keyword to avoid interrupt later login process.

Commit:
login_schema now has correct type keyword.
remove double json dump

Commit:
Add test cases for auth_handler

Commit:
update app.py

Commit:
rename login_handler.py to auth_handler.py;
implemented login/logout/refresh;

add new error code for unauthorized JWT access

Commit:
Docstring added.
file reformated
User DNE testcase for auth_handler added

cpascale43 committed:
starter for /auth route

cpascale43 committed:
adds auth handling capabilities

skassam21 reviewed (approved):
gitignore /venv folder and then you can merge it
remove all the venv folder from github before merging - https://stackoverflow.com/questions/13541615/how-to-remove-files-that-are-listed-in-the-gitignore-but-still-on-the-repositor

cpascale43 committed:
deletes server folder from git
